src/containers/RepositoriesContainer/RepositoriesContainer.js

```diff
@@ -0,0 +1,45 @@
+import React, { PropTypes, Component } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+// Actions
+import * as actionCreators from './RepositoriesDucks.js';
+
+// Components
+import RepoList from '../../components/RepoList/';
+
+const mapStateToProps = (state) => ({
+  isLoading: state.repositories.isLoading,
+  repos: state.repositories.list
+});
+
+const mapDispatchToProps = (dispatch) => bindActionCreators(actionCreators, dispatch);
+
+class RepositoriesContainer extends Component {
+
+  static propTypes = {
+    isLoading: PropTypes.bool.isRequired,
+    repos: PropTypes.array,
+    load: PropTypes.func.isRequired
+  };
+
+  constructor (props) {
+    super(props);
+
+    if (typeof props.repos === 'undefined') {
+      props.load();
+    }
+  }
+
+  render () {
+    const { isLoading, repos } = this.props;
+
+    if (repos && repos.length > 0) {
+      return <RepoList repos={repos} />;
+    }
+
+    return isLoading ? <p>Loading...</p> : <p>No repositories to display.</p>;
```

Comment:
Going to reply to both comments above here because I'm not sure yet. In theory, yes a container should hold virtually no display code.

However, I do need a component which works out if a user should be shown a loading spinner, data etc. The container is definitely the right place to kick off loading that data, so I let this component control displaying that or not as well.

Can't see why that couldn't be moved into RepoList as well though to be fair. I'll move it now and see. Definitely keen to move as much JSX as possible out of the containers! :)